sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -105,6 +105,13 @@ object PredicateSubquery {
       case _ => false
     }.isDefined
   }
+  def hasNullAwarePredicate(e: Expression): Boolean = {
+    e.find(_.isInstanceOf[Not]).isDefined &&
```

Comment:
You are looking for a NOT expression with a nested NULL-aware predicate here right? This will also find other results e.g.: an OR with an NOT on the left hand side and a NULL-aware predicate on the right hand.